Add per-guild settings

amberstarlight opened the pull request:
This PR refactors Cutiebot's configuration. Now taking advantage of the database, we store settings per-guild instead of in `config.json`, which now only holds the bot token. All commands have been given a pass over and everything uses embeds, which greatly separates the bot's messages from users.

amberstarlight committed:
Refactor reminder system

- Move database functions to helper module
- Refactor remind command to use helper module
- Refactor reminders command to use helper module
- Refactor ready event to use helper module
- Add checks for database in index.js

amberstarlight committed:
Add per-guild settings

- Add settings utility module
- Add settings command
- Add guildCreate and guildDelete event
- Create embed helper module
- Update Uncaught Promise Rejection logging to log the stack trace
- Add settings logging in the ready event
- Update commands to use guild settings and embeds
- Fix kick and ban event commands to use GuildMember object rather than a User object
- Update eval and reboot commands to return silently if not run by bot owner (for security)
- Update test suite to use Node environment and embed module

mini-ninja-64 commented on commands/reminders.js:
```diff
@@ -10,47 +10,44 @@ module.exports = {
+		const embed = new Discord.MessageEmbed()
+			.setColor("#36393f");
 		// select all from reminders
-		const reminders = db.prepare("SELECT * from reminders WHERE user_id = (?)").all(message.author.id);
+		// const reminders = db.prepare("SELECT * from reminders WHERE user_id = (?)").all(message.author.id);
```
unnessecary old comments

mini-ninja-64 commented on commands/reminders.js:
```diff
@@ -10,47 +10,44 @@ module.exports = {
 	cooldown: 5,
 	guildOnly: false,
 	execute(message, args) {
+		const embed = new Discord.MessageEmbed()
+			.setColor("#36393f");
```
This file should be updated to use the embed util

mini-ninja-64 commented on events/guildMemberAdd.js:
```diff
@@ -1,6 +1,13 @@
+
+	const welcomeChannel = member.guild.channels.cache.get(guildSettings.welcome_channel_id);
+
 	welcomeChannel.send(
```
Should these be embedified

mini-ninja-64 commented on index.js:
```diff
@@ -3,11 +3,13 @@ loggerInit();
 log.time("startup");
 log.info("Starting Cutiebot!");
 
+// import discord.js, token, and set up client
```
Probably redundant comment

mini-ninja-64 reviewed (changes requested):
lookin nice, couple changes

amberstarlight committed:
Apply review suggestions

- Fix bug in reminders module with use of 'new' keyword
- Remove redundant comments
- Update guildMember events to use embeds

mini-ninja-64 reviewed (approved):
Looks good, nice catch with the new Boolean vs Boolean casting
